src/stan/mcmc/chains.hpp

```diff
@@ -292,6 +291,91 @@ namespace stan {
         return ess;
       }
 
+      /**
+       * Returns the effective sample size for the specified parameter
+       * across all kept samples.
+       *
+       * The implementation is close to the effective sample size
+       * description in BDA3 (p. 286-287).  See more details in Stan
```

Comment:
"BDA3" isn't an adequate citation.  Not everyone's familiar with Andrew's cutesy abbreviations for things.  I would just point to the Stan manual here.